code/lib/blocks/src/blocks/Source.tsx

```diff
@@ -69,16 +69,13 @@ const getSnippet = ({
   typeFromProps: SourceType;
   transformFromProps?: SourceProps['transform'];
 }): string => {
-  const { __isArgsStory: isArgsStory } = storyContext.parameters;
-  const sourceParameters = (storyContext.parameters.docs?.source || {}) as SourceParameters;
+  const parameters = storyContext.parameters ?? {};
+  const { __isArgsStory: isArgsStory } = parameters;
+  const [transformedCode, setTransformedCode] = useState('Loading...');
```

Comment:
1. Maybe `'Transforming...'` instead of `'Loading...'`?
2. the default could also just be `code` if we detect that there is no transformer, and `'Transforming...'` if there is?
3. The default could also just always be `code`, to be replaced by the transformed code afterwards.

Reply:
> the default could also just be code if we detect that there is no transformer, and 'Transforming...' if there is?
It is a bit of a spaghetti code, but at the bottom of the hook, `transformedCode` is not used if `code` is defined. I don't prefer to set the default to `code` because then you might have a flickering between two codes, and you might not understand why it was changing. `Transforming...` indicates that the transformation is still in progress and you should wait to interpret the code. `transform` can not just be used to format code, but also to display something completely different (I think we use a very custom transformer for Angular for example).